### ISSUE TITLE
contrastive loss 

### ISSUE THREAD

**Issue by shaoguangcheng** [BVLC/caffe#1675]
I designed a network using caffe to do face verification. I chose CONTRASTIVE LOSS to optimize the net. But I met a problem : 

The loss can not descend to a reasonable level. I have tried many strategies to fix the problem. All failed.   However, if I used the net structure I designed to train the Mnist. It worked very well.

This problem almost drove me mad. Could someone can give some help ? Thank you very much!

**Comment by Prasanna1991** [comment]
I am also using the Siamese model for face verification. Can you post your model?? And most importantly, the exact size of your datasets (and the pairing numbers too) and the solver strategy you used.

**Comment by shaoguangcheng** [comment]
@Prasanna1991  Thank you your reply. Could you leave me your email address and I write to you about the details

**Comment by Prasanna1991** [comment]
gyawali.prasanna@gmail.com

**Comment by shelhamer** [comment]
Please ask modeling questions on the [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) mailing list. Thanks!

**Comment by memoarch** [comment]
@shaoguangcheng maybe I can help you on this issue, I achieved some good results on Siamese model based face verification, bnj95051@gmail.com

**Comment by ForeverGaara** [comment]
Hi~ @memoarch and @Prasanna1991, I also have some problem on Siamese model training. I randomly choose 600,000 pairs (300,000 same faces and 300,000 different faces) from CASIA-Webface database as training dataset, I used DeepID model proposed by [1], the final loss is about 0.07, but the face verification accuracy on LFW database is only 72%, the face verification algorithm is Joint Bayesian. Could you give some suggestions to training a good siamese model.

[1]. Sun Y, Wang X, Tang X. Deep learning face representation from predicting 10,000 classes.
